types/yup/index.d.ts

```diff
@@ -518,26 +518,26 @@ interface BaseTestOptions<P extends Record<string, any>, C = object> {
     exclusive?: boolean | undefined;
 }
 
-interface NonExclusiveTestOptions<P extends Record<string, any>, C> extends BaseTestOptions<P, C> {
+interface NonExclusiveTestOptions<P extends Record<string, any>, C, T> extends BaseTestOptions<P, C, T> {
     exclusive?: false | undefined;
 }
 
-interface ExclusiveTestOptions<P extends Record<string, any>, C> extends BaseTestOptions<P, C> {
+interface ExclusiveTestOptions<P extends Record<string, any>, C, T> extends BaseTestOptions<P, C, T> {
     exclusive: true;
     name: string;
 }
 
-interface NonExclusiveAssertingTestOptions<U, P extends Record<string, any>, C> extends NonExclusiveTestOptions<P, C> {
+interface NonExclusiveAssertingTestOptions<U, P extends Record<string, any>, C> extends NonExclusiveTestOptions<P, C, any> {
     test: AssertingTestFunction<U, C>;
 }
 
-interface ExclusiveAssertingTestOptions<U, P extends Record<string, any>, C> extends ExclusiveTestOptions<P, C> {
+interface ExclusiveAssertingTestOptions<U, P extends Record<string, any>, C> extends ExclusiveTestOptions<P, C, any> {
     test: AssertingTestFunction<U, C>;
 }
 
-export type TestOptions<P extends Record<string, any> = {}, C = object> =
-    | NonExclusiveTestOptions<P, C>
-    | ExclusiveTestOptions<P, C>;
+export type TestOptions<P extends Record<string, any> = {}, C = object, T extends any = {} | null | undefined> =
```

Comment:
`extends any` is weird / borderline meaningless—was there a reason for it? Also, `{} | null | undefined` will be properly interchangeable with `unknown` in TypeScript 4.8, FYI.